packages/components/dropdowns/dropdown-menu/src/dropdown-menu.spec.tsx

```diff
@@ -3,6 +3,10 @@ import SecondaryButton from '@commercetools-uikit/secondary-button';
 import { act, screen, render } from '../../../../../test/test-utils';
 import DropdownMenu from './dropdown-menu';
 
+// Pause test execution to wait for the DOM-operations to finish
+const waitForDOMOperations = async () =>
```

Comment:
Perhaps we can use [the utilities](https://jestjs.io/docs/timer-mocks) Jest already providers for dealing with timers in tests. What do you think?

Reply:
As far as I understood, those utilities are there to avoid actual waiting while running the tests (basically mocking timer-functions). But the `setTimeout` function here is indeed needed to defer the execution of the code following after. The value of `0` milliseconds is making sure, that execution happens immediately after the recent call stack (= calculating the final DOM in this case) is cleared, so there is no actual waiting involved and test-execution won't take any longer than before.

Reply:
Thanks for the explanation, Michael.

I understand the use case and why we need somehow managing the timeout in these tests, what I'm suggesting is that, instead of using custom functions, we can use what the testing library we use already provides.

In this case, using `jest.useFakeTimers()` and `await jest.runAllTimersAsync();` will work and we wouldn't need to introduce custom code in the test suite.

Would that be reasonable for you?

Reply:
I see, thanks for the hint, i would've never guessed that `runAllTimersAsync` is the key. I went through the jest docs but couldn't find anything fitting, and the crucial part that would eventually have made me realize it is, only shows up in my editor as inline-help after putting the function in use:

![Screenshot 2024-06-20 at 09 27 56](https://github.com/commercetools/ui-kit/assets/169909964/cbac836b-c6df-4028-a874-ae9c69914087)

I pushed the changes, please have another look. Thanky you very much!